doc/source/gr_ore_poly.rst

```diff
@@ -0,0 +1,215 @@
+.. _gr-ore-poly:
+
+**gr_ore_poly.h** -- dense univariate Ore polynomials over generic rings
+===============================================================================
+
+A :type:`gr_ore_poly_t` represents a univariate Ore polynomial `L \in R[D]`
+implemented as a dense array of coefficients in a generic ring *R*.
+The choice of Ore operator *D* is stored in the context object
+:type:`gr_ore_poly_ctx_t`.
+
+Most functions are provided in two versions: an underscore method which
+operates directly on pre-allocated arrays of coefficients and generally
+has some restrictions (often requiring the lengths to be nonzero
+and not supporting aliasing of the input and output arrays),
+and a non-underscore method which performs automatic memory
+management and handles degenerate cases.
+
+Ore operators
+--------------------------------------------------------------------------------
+.. type:: ore_poly_which_operator
+
+    Represents one of the following supported Ore operators:
+
+    .. macro:: ORE_OPERATOR_STANDARD_DERIVATIVE
+
+        The standard derivative.
+
+    .. macro:: ORE_OPERATOR_EULER_DERIVATIVE
+
+        The Euler derivative.
+
+.. function:: ore_poly_which_operator ore_poly_which_operator_randtest(flint_rand_t state)
+
+    Return a random Ore operator.
+
+Type compatibility
+-------------------------------------------------------------------------------
+
+The ``gr_ore_poly`` type has the same data layout as the following
```

Comment:
This paragraph doesn't really make sense since the base ring is always supposed to be something like a polynomial ring and not for example ``fmpz`` (even if that is legal). It would be enough to say that the layout is the same as ``gr_poly``, and that ``gr_poly`` methods therefore can be used for linear and container operations on a ``gr_ore_poly`` provided that one is careful about providing the right context object.